lib/matplotlib/quiver.py

```diff
@@ -372,12 +372,13 @@ def _parse_args(*args):
     X, Y, U, V, C = [None] * 5
     args = list(args)
 
-    # The use of atleast_1d allows for handling scalar arguments while also
-    # keeping masked arrays
-    if len(args) == 3 or len(args) == 5:
+    if len(args) == 2 or len(args) == 4:
+        V = np.atleast_1d(args.pop(-1))
+        U = np.atleast_1d(args.pop(-1))
+    elif len(args) == 3 or len(args) == 5:
```

Comment:
I agree that doing the `_check_array` on U, V, C here (where they are defined) is more suitable than leaving them at the end. But I don't think stuffing the `_check_array` and `atleast_1d` together is a good idea, at least, seem less readable to me.